File: Matérias/Aeds II/Beecrowd/Alarme Despertador/Java/Alarme.java
//https://judge.beecrowd.com/pt/problems/view/1103

import java.util.*;
import java.io.*;

public class Alarme
{
    public static void main(String args[])
    {
        Scanner scanf = new Scanner(System.in);

        String input = new String();
        
        for(input = scanf.nextLine(); !(input.equals("0 0 0 0")); input = scanf.nextLine())
        {

            String[] array = input.split(" ");
            int[] time = new int[array.length];

            for (int i = 0; i < time.length; i++) {
                time[i] = Integer.parseInt(array[i]);
            }
            
            int minutosDia1 = (time[0] * 60) + time[1];
            int minutosDia2 = (time[2] * 60) + time[3];

            int resultado = 0;

            if(minutosDia1 < minutosDia2)
            {
                resultado = minutosDia2 - minutosDia1;
            }
            else if(minutosDia1 > minutosDia2)
            {
                resultado = 1440 - minutosDia1 + minutosDia2;
            }

            if(resultado != 0)System.out.println(resultado);
        }

        scanf.close();
    }
}

File: Matérias/Aeds II/Beecrowd/Alarme Despertador/C/Alarme.c
#include <stdlib.h>
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int main()
{
    int hora1, minuto1, hora2, minuto2;
    
    scanf(" %d %d %d %d", &hora1, &minuto1, &hora2, &minuto2);

    getchar();

    while(!(hora1 == 0 && minuto1 == 0 && hora2 == 0 && minuto2 == 0))
    {

        int minutosDia1 = hora1 * 60 + minuto1;
        int minutosDia2 = hora2 * 60 + minuto2;


        int resultado = 0;
        if(minutosDia1 < minutosDia2)
        {
            resultado = minutosDia2 - minutosDia1;
        }
        else if(minutosDia2 < minutosDia1)
        {
            resultado = 1440 - minutosDia1 + minutosDia2;
        }

        printf("%d\n", resultado);

        scanf(" %d %d %d %d", &hora1, &minuto1, &hora2, &minuto2);

        getchar();

    }
    return 0;
}